### PR TITLE
no-copy CLI example

### PR DESCRIPTION
**Pull request by drernie**
# Add sample code to CLI (push)

### CODE REVIEW

**Review by robnewman — CHANGES_REQUESTED** [review]
The file `docs/api-reference/cli.md` should not be deleted as part of the PR

**Review by akarve — APPROVED** [review]

File: docs/api-reference/cli.md [hunk 1]
```diff
@@ -138,7 +138,7 @@ optional arguments:
 ```
 usage: quilt3 push --dir DIR [-h] [--registry REGISTRY] [--dest DEST]
                    [--message MESSAGE] [--meta META] [--workflow WORKFLOW]
-                   [--force] [--dedupe] [--no-copy]
+                   [--force] [--dedupe] [--no-copy] [--detailed_help]
```
Comment (akarve) [inline docs/api-reference/cli.md:141]: Why not just make `-h` work like this instead of adding another command?

Comment (drernie) [reply docs/api-reference/cli.md:141]: `detailed_help` is the existing mechanism for adding documentation to the published web docs.

File: api/python/quilt3/main.py [hunk 2]
```diff
@@ -207,7 +207,31 @@ def _selector_fn_no_copy(*args):
...
+quilt3 push myuser/my_alias --no-copy --dir s3://your-bucket/data --registry s3://my-bucket
+```
+"""
+
+
+def cmd_push(name, dir, registry, dest, message, meta, workflow, force, dedupe, no_copy, detailed_help=False):
+    """
+    Push a Quilt package with `name` to the specified (or default) registry.
+
+    If detailed_help=True, display detailed information about the `quilt3 push` command and then exit
```
Comment (sir-sigurd) [inline api/python/quilt3/main.py:228]:
```suggestion
    If detailed_help=True, display detailed information about the `quilt3 push` command and then exit.
```

File: api/python/quilt3/main.py [hunk 3]
```diff
@@ -479,6 +503,11 @@ def create_parser():
         action="store_true",
         help="Do not copy data. Package manifest entries will reference the data at the original location.",
     )
+    optional_args.add_argument(
+        "--detailed_help",
```
[the remark this answers is not in the document]
Comment (sir-sigurd) [reply api/python/quilt3/main.py:507]: I think you could make requested change given https://github.com/quiltdata/quilt/pull/3439 is merged.

File: api/python/Makefile [hunk 4]
```diff
@@ -4,3 +4,6 @@ test:
 
 install-local:
 	pip install -e .[tests]
+
+testdocs:
+	cd ../../testdocs && poetry install && poetry run pytest --codeblocks ../docs
```
Comment (sir-sigurd) [inline api/python/Makefile:9]: Why not have a newline here?
```suggestion
	cd ../../testdocs && poetry install && poetry run pytest --codeblocks ../docs

```

**Review by sir-sigurd — CHANGES_REQUESTED** [review]

**Comment by drernie** [comment]
Don't care enough anymore.

**Pull request closed without merging**